test/passes/Oz_fuzz-exec_all-features.wast

```diff
@@ -141,4 +141,42 @@
    )
   )
  )
+ (func "br_on_cast"
+  (local $any anyref)
+  ;; create a simple $struct, store it in an anyref
+  (local.set $any
+   (struct.new_default_with_rtt $struct (rtt.canon $struct))
+  )
+  (drop
+   (block $block (result ($ref $struct))
+    (drop
+     (block $extendedblock (result (ref $extendedstruct))
+      (drop
+       ;; second, try to cast our simple $struct to what it is, which will work
+       (br_on_cast $block $struct
+        ;; first, try to cast our simple $struct to an extended, which will fail
+        (br_on_cast $extendedblock $extendedstruct
+         (local.get $any) (rtt.canon $extendedstruct)
```

Comment:
Isn't this `extendedstruct`...?

Reply:
Do you mean `$any`? It's assigned a `$struct` on line 148. So casting it to an `$extendedstruct` fails.

Reply:
Oh right, I was confused the heap type annotation with the argument.. Sorry.